<a id= "flow_control"> </a>
## 1. Python Flow Control

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>1. Write a code to print table of 5 using assignment operand +=.</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [7]:
l1 = [2,4,9]
l2 = [3,5,7]
[(l1,l2) for a in l1 for b in l2 if l1>l2]

[]

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>2. Write a program for generating a fibonacci series starting with 0 and 1 for the next 10 values using a while loop. </b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [2]:
# type your code here
n1=0
n2=1
n=0
list=[]

while n < 13:
    list.append(n1)
    nth=n1+n2
    n1=n2
    n2=nth
    n=n+1
print(list)

    

[0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144]


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>3. Write a program to check whether a number is greater than the other using if statement (take the input from user).</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [3]:
# type your code here
n=int(input('first num '))
n1=int(input('sec num '))
if n>n1:
    print("greater num is ",n)
else:
    print("greater num is ",n1)

first num 6
sec num 7
greater num is  7


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>4. Write a code to check whether a number is divisible by 7 or not (take the input from user). </b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [4]:
# type your code here
n=int(input('enter a num'))
if n%7==0:
    print("the num is divisible by 7")
else:
    print("not divisible by 7")

enter a num5
not divisible by 7


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>5. Write a code to find factorial of a number (take the input from user). </b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [5]:
# type your code here
n=int(input('enter a num'))
res=1
for i in range(1,n+1):
    res= res*i
print(res)

enter a num5
120


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>6. Write a program to check whether a number is prime or not (take the input from user).</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [6]:
# type your code here
n=int(input('enter a num'))
for i in range(2,n):
    if n%i==0:
        print('not prime')
        break
else:
    print('prime')
        

enter a num29
prime


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>7. Write a program to check whether two numbers are amicable or not (take the input from user).</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [6]:
# type your code here
x=int(input('Enter first number : '))  
y=int(input('Enter second number : '))  
sum1=0  
sum2=0  
for i in range(1,x):  
    if x%i==0:  
        sum1+=i  
for j in range(1,y):  
    if y%j==0:  
        sum2+=j  
if(sum1==y and sum2==x):  
    print('Given numbers are Amicable!')  
else:  
    print('Given numbers are not Amicable!')  


Enter first number : 220
Enter second number : 284
Given numbers are Amicable!


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>8. Reverse string using a for loop (take the input from user).</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [7]:
# type your code here
string=input("enter a string")
res=""
for i in string:
    res=i + res
print(res)

enter a stringvs
sv


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>9. Write a code to find the average of given numbers (take the input from user).</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [14]:
# type your code here

n=input('enter numbers').split()
sum=0
for i in n:
    sum=sum+int(i)
print("avg is : " , sum/len(n))
    




enter numbers4 4 5 5
avg is :  4.5


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>10. Write a program to find the area of a circle for a given radius (take the input from user).</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [ ]:
# type your code here
n=float(input('enter the radius'))
area= 3.14*n*n
print(area)

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>11. Write a code to find the simple interest (take the input from user).</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [16]:
# type your code here
p,r,t=input('enter numbers').split()
si=(int(p)*int(r)*int(t)/(100))
print(si)



enter numbers5 1 2
0.1


<a id= "lc"> </a>
## 2. List Comprehension

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>12. Use list comprehension to obtain the squre root of first 10 natural numbers.</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [ ]:
# type your code here
lst=[i**0.5 for i in range(1,11)]
lst

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>13. Use list comprehension to find the used vowels in a given sentence and/or word (take the input from user). </b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [17]:
# type your code here
string=input('enter a string')
lst=['a','e','i','o','u']
for i in string:
    if i in lst:
        print(i)
        pass
        

enter a stringaeioub
a
e
i
o
u


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>14. Create a dictionary and access it values using a condition on its key. The data is given below. Let the condidtion on the key is that it should be a even number.</b>
                </font>
            </div>
       </td>
    </tr>
</table>


<table>
<tr>
    <th>Key</th>
    <td>1</td>
    <td>2</td>
    <td>3</td>
    <td>4</td> 
    <td>5</td> 
</tr>
<tr>
    <th>Name</th>
    <td>Aman</td>
    <td>Mohit</td>
    <td>Guari</td>
    <td>Imran</td>
    <td>Roma</td
     </tr>
<tr>
    <th>Marks</th>
    <td>24</td>
    <td>25</td>
    <td>26</td>
    <td>24</td>
    <td>27</td>
   </tr>
</table>

In [50]:
letters = ["beach", "car"]
funified = list(map(lambda word: f"{word} is fun!", letters))
print(funified)

['beach is fun!', 'car is fun!']


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>15. Use list comprehension to find even and odd numbers from first 20 whole numbers.</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [26]:
# type your code here
lst=[  "even" if i%2==0 else "odd" for i in range(1,21)     ]
lst

['odd',
 'even',
 'odd',
 'even',
 'odd',
 'even',
 'odd',
 'even',
 'odd',
 'even',
 'odd',
 'even',
 'odd',
 'even',
 'odd',
 'even',
 'odd',
 'even',
 'odd',
 'even']

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>16. Use list comprehension to print numbers divisible  by 2 and 3 in between 1 and 100. </b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [22]:
# type your code here
lst=[x for x in range(1,101) if x%2==0 and x%3==0]
lst

[6, 12, 18, 24, 30, 36, 42, 48, 54, 60, 66, 72, 78, 84, 90, 96]

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>17. Use list comprehension to create a dictionary such that its keys are numbers from 1 to 10 and values are the corresponding to that key are its cube.</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [1]:
# type your code here
dic={x: x**3 for x in range(1,11)}
dic

{1: 1, 2: 8, 3: 27, 4: 64, 5: 125, 6: 216, 7: 343, 8: 512, 9: 729, 10: 1000}

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>18. Use list comprehension to extract numbers from a string </b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [5]:
# type your code here
n=input('enter a string with numbers')
lst=[int(x) for x in n if x.isnumeric()]
lst

enter a string with numbersabfbb2938


[2, 9, 3, 8]

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>19. Use list comprehension to print table of 11 to 20 as shown below.</b>
                </font>
            </div>
        </td>
    </tr>
</table>

<table>
<tr>
    <td>11</td><td>22</td><td>33</td><td>44</td><td>55</td><td>66</td><td>77</td><td>88</td><td>99</td><td>110</td> 
</tr>
    
<tr>
    <td>12</td><td>24</td><td>36</td><td>48</td><td>60</td><td>72</td><td>84</td><td>96</td><td>108</td><td>120</td> 
</tr>

<tr>
    <td>13</td><td>36</td><td>39</td><td>52</td><td>65</td><td>78</td><td>91</td><td>104</td><td>117</td><td>130</td> 
</tr> 

<tr>
     <td>14</td><td>28</td><td>42</td><td>56</td><td>70</td> <td>84</td><td>98</td><td>112</td><td>126</td><td>140</td> 
</tr>

<tr>
     <td>15</td><td>30</td><td>45</td><td>60</td><td>75</td><td>90</td><td>105</td><td>120</td><td>135</td><td>150</td> 
</tr>

<tr>
     <td>16</td><td>32</td><td>48</td><td>64</td><td>80</td><td>96</td><td>112</td><td>128</td><td>144</td><td>160</td> 
</tr> 

<tr>
     <td>17</td><td>34</td><td>51</td><td>68</td><td>85</td><td>102</td><td>119</td><td>136</td><td>153</td><td>170</td> 
</tr> 

<tr>
     <td>18</td><td>36</td><td>54</td><td>72</td><td>90</td> <td>108</td><td>126</td><td>144</td><td>162</td><td>180</td> 
</tr>

<tr>
 <td>19</td><td>38</td><td>57</td><td>76</td><td>95</td><td>114</td><td>133</td><td>152</td><td>171</td><td>190</td> 
</tr>

<tr>
     <td>20</td><td>40</td><td>60</td><td>80</td><td>100</td><td>120</td><td>140</td><td>160</td><td>180</td><td>200</td> 
<tr>    
    
</table>

In [18]:

# Function to calculate sum of all elements in matrix
# sum(arr) is a python inbuilt function which calculates
# sum of each element in a iterable ( array, list etc ).
# map(sum,arr) applies a given function to each item of
# an iterable and returns a list of the results.
def findSum(arr):
 
    # inner map function applies inbuilt function
    # sum on each row of matrix arr and returns
    # list of sum of elements of each row
    lst=map(sum,arr)
    return list(lst)
 
# Driver function
if __name__ == "__main__":
    arr = [[1, 2, 3], [4, 5, 6], [2, 1, 2]]
    print ("Sum = ",findSum(arr))

Sum =  [6, 15, 5]


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>20. Use list comprehension to find transpose of the matrix given below.</b>
                </font>
            </div>
        </td>
    </tr>
</table>

<table>
    <tr>
        <td>1</td>
        <td>2</td>
    </tr>
    <tr>
        <td>3</td>
        <td>4</td>
    </tr>
    <tr>
        <td>5</td>
        <td>6</td>
    </tr>
    <tr>
        <td>7</td>
        <td>8</td>
    </tr>
    </table>

In [20]:
# type your code here
m = [[1,2],[3,4],[5,6],[7,8]]
for row in m :
    print(row)
rez = [[m[j][i] for j in range(4)] for i in range(2)]
print("\n")
for row in rez:
    print(row)
        

[1, 2]
[3, 4]
[5, 6]
[7, 8]


[1, 3, 5, 7]
[2, 4, 6, 8]


<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>21.  Assume a mother playing a "word guessing game  2 times" with children. She chooses 8 words relating to recent conversation  and write sets of clues to help students. Condition that children should not use same word again. children get 5 chances to guess the four letters word given below.
 
 Create a word set=["milk", "race", "rice", "race", "hide", "tree","jump","bike"] 
 
 like and take user input display like "Let's play a word guessing game . The word has 4 letters in it. You get 5 guesses. Guess a letter " if you guess correct 2 time it will display. If children found answer with 2 option display wow
 
 Great Guess! for 1st time
 
 Great Guess! for 2nd time
 
 Thanks for playing! you won!
 
 
 Exiting
 
 If your guess wrongly your code will ask 5 times to guess the word as shown in the below example
 
 Sorry, try again. was
 
 Sorry, try again. the
 
 Sorry, try again. are
 
 Sorry, try again. the
 
 Sorry, try again. are
 
 if you did not find 2 option it return you lose.
 
 
 make sure that program takes only smaller case values..</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [8]:
import random
word_set=["milk", "race", "rice", "race", "hide", "tree","jump","bike"]

print("Let's play a word guessing game.")
print("The word has 4 letters in it.")
print("You get 5 guesses.", end='\n\n')
word = random.choice(word_set)
correct_guess = 0
guess = 1

while guess <= 5: 
    letter = input("Guess a letter:")
    letter = letter.lower()
    if letter in word:
        correct_guess += 1
        print("Great Guess!")
        print("The number of correct guesses:", correct_guess)
        if correct_guess == 2:
            print("Thanks for playing! You won!")
            break
    else:
        print("Sorry, try again")

    print("Remaining attempts:", 5 - guess)
    print()
    guess+=1

else:
    print("Thanks for playing! You Lost!")

Let's play a word guessing game.
The word has 4 letters in it.
You get 5 guesses.

Guess a letter:i
Great Guess!
The number of correct guesses: 1
Remaining attempts: 4

Guess a letter:k
Great Guess!
The number of correct guesses: 2
Thanks for playing! You won!
